From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Helvetica Volubi <suisuroru@blue-millennium.fun>
Date: Wed, 22 Apr 2026 12:59:08 +0800
Subject: [PATCH] Leaves: Leaves plugin

Co-authored by: MC_XiaoHei <xiaohei.xor7@outlook.com>
As part of: Leaves (https://github.com/HyacinthHaru/Leaves/blob/90080d238e2fcfc14a9293150dc351a4f0f696e6/leaves-api/paper-patches/features/0009-Leaves-Plugin.patch)
Licensed under: GPL-3.0 (https://www.gnu.org/licenses/gpl-3.0.html)

diff --git a/src/main/java/org/bukkit/Bukkit.java b/src/main/java/org/bukkit/Bukkit.java
index 550c7eef69acde07c05b0cb75bf413186c09a26a..8e96326c55510642df01ef15e1a3edb1e81176cb 100644
--- a/src/main/java/org/bukkit/Bukkit.java
+++ b/src/main/java/org/bukkit/Bukkit.java
@@ -3042,4 +3042,10 @@ public final class Bukkit {
     public static void restart() {
         server.restart();
     }
+
+    // Leaves start - Feature API
+    public static org.leavesmc.leaves.plugin.FeatureManager getFeatureManager() {
+        return server.getFeatureManager();
+    }
+    // Leaves end - Feature API
 }
diff --git a/src/main/java/org/bukkit/Server.java b/src/main/java/org/bukkit/Server.java
index 23901bb9e01a90dc5f12e458b7188d51e91022b4..473b0e02c003d5317ef798c320e5c845124f4e12 100644
--- a/src/main/java/org/bukkit/Server.java
+++ b/src/main/java/org/bukkit/Server.java
@@ -2780,6 +2780,9 @@ public interface Server extends PluginMessageRecipient, net.kyori.adventure.audi
      */
     void allowPausing(@NotNull org.bukkit.plugin.Plugin plugin, boolean value);
     // Paper end - API to check if the server is sleeping
+    // Leaves start - Feature API
+    org.leavesmc.leaves.plugin.FeatureManager getFeatureManager();
+    // Leaves End - Feature API
     // Folia start - region TPS API
     /**
      * Gets the TPS from the region which owns the specified location, or {@code null} if no region owns
